src/java.base/share/classes/java/lang/Integer.java

```diff
@@ -514,14 +524,16 @@ static int getChars(int i, int index, byte[] buf) {
             q = i / 100;
             r = (q * 100) - i;
             i = q;
-            buf[--charPos] = DigitOnes[r];
-            buf[--charPos] = DigitTens[r];
+            charPos -= 2;
+            UNSAFE.putShortUnaligned(buf, Unsafe.ARRAY_BYTE_BASE_OFFSET + charPos, PACKED_DIGITS[r], false);
```

Comment:
When switching to use Unsafe, `getChars` should do the array bounds check in the loop of the store index.

Reply:
The value range of the r variable is 0-99, and the length of PACKED_DIGITS is 100, There is no need to check the array boundary here.

Reply:
I think he means to check the `charPos` to ensure it is not out of bounds.

Reply:
Oh, I see, charPos needs to do bound checks,  I added assert with reference to the implementation of StringUTF16#putChar, is this safe enough?

Reply:
I think we might need to look into the code around to ensure user input cannot cause charPos to go out of bounds. If charPos is not touched by user code at all, I think assert suffices (which are enabled via -esa and is thus enabled in the jtreg test suite)